src/data/sym/basic.lean

```diff
@@ -150,4 +155,100 @@ instance inhabited_sym' [inhabited α] (n : ℕ) : inhabited (sym' α n) :=
 
 end inhabited
 
+instance has_zero : has_zero (sym α 0) := ⟨⟨0, rfl⟩⟩
+instance has_emptyc : has_emptyc (sym α 0) := ⟨0⟩
+
+instance subsingleton (n : ℕ) [subsingleton α] : subsingleton (sym α n) :=
```

Comment:
Can you also add
```lean
instance unique_zero (n : ℕ) : unique (sym α 0) :=
```
If you have that available, you can reuse it to prove the `n=0` case of the `subsingleton` lemma I'm commenting on.

Reply:
```lean
lemma eq_nil_of_card_zero (s : sym α 0) : s = nil :=
begin
  obtain ⟨a, h⟩ := s,
  rw multiset.card_eq_zero at h,
  subst h,
  refl,
end

instance unique_zero : unique (sym α 0) :=
⟨⟨nil⟩, eq_nil_of_card_zero⟩
```

Reply:
And a suggestion for `subsingleton`:
```lean
lemma repeat_succ {a : α} {n : ℕ} : repeat a n.succ = a :: repeat a n := rfl

lemma exists_eq_cons_of_succ (s : sym α n.succ) : ∃ (a : α) (s' : sym α n), s = a :: s' :=
begin
  classical,
  obtain ⟨m, h⟩ := s,
  have : 0 < multiset.card m,
  { rw h,
    exact nat.succ_pos n, },
  rw multiset.card_pos_iff_exists_mem at this,
  obtain ⟨a, ha⟩ := this,
  use [a, m.erase a],
  { rw [multiset.card_erase_of_mem ha, h],
    refl, },
  { rw cons,
    congr,
    rw multiset.cons_erase ha, },
end

lemma eq_repeat_of_subsingleton [subsingleton α] (a : α) {n : ℕ} (s : sym α n) : s = repeat a n :=
begin
  induction n with n ih,
  { rw eq_nil_of_card_zero s,
    refl, },
  { obtain ⟨b, s', rfl⟩ := exists_eq_cons_of_succ s,
    rw [repeat_succ, ih s'],
    congr, },
end

instance subsingleton (n : ℕ) [subsingleton α] : subsingleton (sym α n) :=
⟨begin
  cases n,
  { simp, },
  { intros s s',
    obtain ⟨b, s, rfl⟩ := exists_eq_cons_of_succ s,
    rw [eq_repeat_of_subsingleton b s', eq_repeat_of_subsingleton b s, repeat_succ], },
end⟩
```